download/internalRewriteDownload/test/InternalRewriteDownloadTest.groovy

```diff
@@ -0,0 +1,56 @@
+import spock.lang.Specification
+import static org.jfrog.artifactory.client.ArtifactoryClient.create
+import org.jfrog.artifactory.client.model.repository.settings.impl.GenericRepositorySettingsImpl
+
+
+class InternalRewriteDownloadTest extends Specification {
+
+    def 'test'() {
+        setup:
+        def baseurl = 'http://localhost:8088/artifactory'
+        def artifactory = create(baseurl, 'admin', 'password')
+         
+
+        
+        //building dist-local repo
+        def builder = artifactory.repositories().builders()
+        def local = builder.localRepositoryBuilder().key('dist-local')
+            .repositorySettings(new GenericRepositorySettingsImpl()).build()
+            artifactory.repositories().create(0, local)
+
+        def repo = artifactory.repository("dist-local")
+
+
+        def myVersion = "VERSION" 
+        myVersion = "latest"
+        
+        //uploading first file
+        def path ="$myVersion/test1.txt"
+        def file = new File('./src/test/groovy/InternalRewriteDownloadTest/test1.txt')
+        file.write "$myVersion"        
+        repo.upload(path,file).doUpload();
```

Comment:
@yamaguchim Instead of writing the artifact's content to a file before uploading it, you could just convert the desired content to a stream of bytes and upload it as showed in the snippet below:
```groovy
repo.upload(path, new ByteArrayInputStream('content'.bytes)).doUpload()
```